edm4hep/CMakeLists.txt

```diff
@@ -7,6 +7,12 @@ PODIO_ADD_DATAMODEL_CORE_LIB(edm4hep "${headers}" "${sources}")
 add_library(EDM4HEP::edm4hep ALIAS edm4hep)
 set_target_properties(edm4hep PROPERTIES ALIAS_GLOBAL true)
 
+file(GLOB_RECURSE top_headers ${PROJECT_SOURCE_DIR}/include/*.h)
+target_sources(edm4hep PUBLIC FILE_SET headers TYPE HEADERS BASE_DIRS ${PROJECT_SOURCE_DIR}/include FILES ${top_headers})
```

Comment:
Are these the generated headers that we are re-discovering again with globbing, or are these other headers?

Reply:
Just the ones in `/include`; instead of installing them in the top CMakeLists.txt it is done here in L38 together with the edm4hep library

Reply:
Should the generated headers also go into the public headers then?

Reply:
The `FILE_SET` seems to be cmake newest take on headers management. 
Alternatively the things could be done the same way as in podio (works with older cmake), namely with the `PUBLIC_HEADER` property:
```cmake
set_property(
    TARGET edm4hep
    APPEND PROPERTY
    PUBLIC_HEADER ${top_headers}
)
```

The `PUBLIC_HEADER` has to be installed in the L37 anyway since this is how podio handles headers (generated headers added to the property by `PODIO_ADD_DATAMODEL_CORE_LIB`).

The advantage of `FILE_SET` over `PUBLIC_HEADER` is preserving the hierarchy in the include directories instead of flattening it. In case of this module the hierarchy is flat anyway.

Reply:
> Should the generated headers also go into the public headers then?

Must? Nope. Would it be nice? Maybe and some cmake code can be simplified; I'll have a look but in a different PR since it's not needed and I think the change comes from or also in the podio macros in that case

Reply:
Would probably be a podio feature request in any case.